src/eliminate_contiguous.cpp

```diff
@@ -9,19 +9,61 @@
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
-bool try_compute_shape(const operation& op, const std::vector<instruction_ref>& args)
+static bool try_compute_shape(instruction_ref ins, const std::vector<shape>& inputs)
 {
     try
     {
-        compute_shape(op, args);
+        shape new_shape = ins->get_operator().compute_shape(inputs);
+        // If the output shape is a standard shape, no need to try its output
+        if(new_shape.standard())
+        {
+            return true;
+        }
+
+        // if no changes for the shape, the contiguous can also be removed
+        if(new_shape == ins->get_shape())
+        {
+            return true;
+        }
+
+        auto outputs = ins->outputs();
+        // If the current instruction has no output, it means it is the last
+        // instruction and generates a non-standard output. But for unary
+        // and binary operators, we can still remove it and reshape the output
+        // to be standard since these operator can handle non-standard inputs
+        if(outputs.empty())
+        {
+            return true;
+        }
+
+        for(auto output : outputs)
+        {
+            auto args = output->inputs();
+            std::vector<shape> input_shapes(args.size());
+            std::transform(args.begin(), args.end(), input_shapes.begin(), [&](auto& arg) {
+                return (arg == ins) ? new_shape : arg->get_shape();
+            });
```

Comment:
I think it would be more readable to use `std::replace_if` instead of transform.

Reply:
Do you mean doing it like: input_shapes = to_shape(args);
std::replace(input_shapes.begin(), input_shapes.end(), [](auto &s) { return s == ins->get_shape(); }, new_shape);